cmd/chain-maind/app/app.go

```diff
@@ -142,11 +143,20 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 		config.SetRoot(clientCtx.HomeDir)
 		path := config.GenesisFile()
 
-		file, err := os.OpenFile(path, os.O_RDWR, 0644)
+		cleanedPath := filepath.Clean(path)
+		// nolint: gosec
+		file, err := os.OpenFile(cleanedPath, os.O_RDWR, 0600)
 		if err != nil {
 			return err
 		}
-		defer file.Close()
+
+		defer func() {
+			cerr := file.Close()
+			if err == nil {
+				err = cerr
+			}
```

Comment:
and if err is not nil, it'll be lost? I'm not sure of the exact defer semantics, but perhaps logging the error (if cerr is not nil) won't hurt?

Reply:
I think assign to err at here don't have any effect, it's in panic handling process here, the err is not consumed by anybody.

Reply:
yes, i changed to display the log